pkg/operator/controller/ingress/deployment.go

```diff
@@ -136,6 +139,28 @@ func HTTP2IsEnabled(ic *operatorv1.IngressController, ingressConfig *configv1.In
 	return configHasHTTP2Enabled
 }
 
+// HardStopAfterIsEnabledByAnnotation returns true if the map m has
+// the key RouterHardStopAfterEnvName and its value is not the empty
+// string ("").
+func HardStopAfterIsEnabledByAnnotation(m map[string]string) (bool, string) {
```

Comment:
No need to export this, is there?  Same for `HardStopAfterIsEnabled`.  (But `HTTP2IsEnabledByAnnotation` is already public, and anyway it's a minor thing, so I won't object if you prefer to keep this as-is.)
```suggestion
// hardStopAfterIsEnabledByAnnotation returns true if the map m has
// the key RouterHardStopAfterEnvName and its value is not the empty
// string ("").
func hardStopAfterIsEnabledByAnnotation(m map[string]string) (bool, string) {
```

Reply:
> No need to export this, is there? Same for `HardStopAfterIsEnabled`. (But `HTTP2IsEnabledByAnnotation` is already public, and anyway it's a minor thing, so I won't object if you prefer to keep this as-is.)

Currently used by e2e tests (not pushed).  If I don't need it then I'll change the scope.